pkg/virt-api/rest/kubeproxy.go

```diff
@@ -138,39 +142,41 @@ func GenericResourceProxy(ws *restful.WebService, ctx context.Context, gvr schem
 			To(endpoints.MakeGoRestfulWrapper(getListAllNamespaces)).Writes(listExample).
 			Doc("Get a list of all "+objKind+" objects.").
 			Returns(http.StatusOK, "OK", listExample).
-			Returns(http.StatusNotFound, "Not Found", nil), ws,
+			Returns(http.StatusUnauthorized, "Unauthorized", nil), ws,
 	))
 
 	ws.Route(
 		ws.PATCH(ResourcePath(gvr)).
-			Produces(mime.MIME_JSON_PATCH).
-			Operation("updateNamespaced"+objKind).
-			To(endpoints.MakeGoRestfulWrapper(patch)).Writes(objExample).
+			Consumes(mime.MIME_JSON_PATCH, mime.MIME_MERGE_PATCH).
+			Produces(mime.MIME_JSON).
+			Operation("patchNamespaced"+objKind).
+			To(endpoints.MakeGoRestfulWrapper(patch)).
+			Writes(objExample).Reads(metav1.Patch{}).
 			Doc("Patch a "+objKind+" object.").
-			Returns(http.StatusOK, "Patched", objExample).
-			Returns(http.StatusNotFound, "Not Found", nil),
+			Returns(http.StatusOK, "OK", objExample).
+			Returns(http.StatusUnauthorized, "Unauthorized", nil),
 	)
 
 	// TODO, implement watch. For now it is here to provide swagger doc only
 	ws.Route(addWatchGetListParams(
 		ws.GET("/watch/"+gvr.Resource).
 			Produces(mime.MIME_JSON).
 			Operation("watch"+objKind+"ListForAllNamespaces").
-			To(NotImplementedYet).Writes(listExample).
+			To(NotImplementedYet).Writes(metav1.WatchEvent{}).
 			Doc("Watch a "+objKind+"List object.").
-			Returns(http.StatusOK, "OK", listExample).
-			Returns(http.StatusNotFound, "Not Found", nil), ws,
```

Comment:
Are you sure that the API can never return 404, or is that already globally added?

Reply:
I think API can return 404 for sure, but when I was listing k8s spec I can not see 404 be defined there ...
I don't have any reason to remove it except the fact that it is missing in k8s.
If you want I can put it back.